spec/integration/prometheus_spec.rb

```diff
@@ -0,0 +1,46 @@
+require_relative '../vcr/vcr_setup'
+require_relative '../spec_helper'
+
+module Hawkular::Prometheus::RSpec
+  HAWKULAR_BASE = 'http://localhost:8080/'
+  creds = {}
+  options = {}
+
+  describe 'Prometheus/Queries' do
+    let(:cassette_name) do |example|
+      description = example.description
+      description
+    end
+
+    around(:each) do |example|
+      record('Prometheus/Queries', credentials, cassette_name, example: example)
+    end
+
+    before(:each) do
+      @client = Hawkular::Prometheus::Client.new(HAWKULAR_BASE, creds, options)
+    end
+    it 'Should Fetch Metrics' do
+      metrics = [
+        Hawkular::Inventory::Metric.new('displayName' => 'Heap Used',
+                                        'family' => 'jvm_memory_bytes_used',
+                                        'unit' => 'BYTES',
+                                        'expression' => 'jvm_memory_bytes_used{area="heap",feed_id="attilan"}',
+                                        'labels' => { area: 'heap', feed_id: 'attilan' }),
+        Hawkular::Inventory::Metric.new('displayName' => 'NonHeap Used',
+                                        'family' => 'jvm_memory_bytes_used',
+                                        'unit' => 'BYTES',
+                                        'expression' => 'jvm_memory_bytes_used{area="nonheap",feed_id="attilan"}',
+                                        'labels' => { area: 'nonheap', feed_id: 'attilan' })
+      ]
+
+      results = @client.query_range(metrics: metrics,
+                                    starts: '2017-11-21T09:00:00Z',
+                                    ends: '2017-11-21T11:00:00Z',
+                                    step: '5s')
+
+      expect(results.map { |r| r['metric'].name }).to include('Heap Used', 'NonHeap Used')
```

Comment:
Here I'd test each one of them individually, like so:

```ruby
first = results.first
second = results.last

expect(first['metric'].name).to eq 'Heap Used'
expect(second['metric'].name).to eq 'NonHeap Used'
```

The less processing we do on each test, the better.

Reply:
+1 
I like this approach, I will change it, thx.